lib/shark_client.js

```diff
@@ -129,6 +127,21 @@ function _request(opts, cb) {
     }
 }
 
+/*
+ * Use information about the object to craft the storage path.
+ */
+function storagePath(opts) {
+    var storageLayoutVersion = '/v' + opts.storageLayoutVersion + '/';
+    var owner = opts.creator || opts.owner;
+    var bucketId = opts.bucketId;
+    var objectNameHash = opts.objectNameHash;
+    var objectId = opts.objectId;
+    var objectIdAndHash = objectId + ',' + objectNameHash;
+    var objectIdPrefix = objectId.substr(0, 2);
+
+    return (path.join(storageLayoutVersion, owner, bucketId, objectIdPrefix,
```

Comment:
Perhaps explicitly use `'/'` for joining rather than `path.join` which is about OS-indep path separators.